packages/admin/blocks-admin/src/blocks/factories/createCompositeBlock.tsx

```diff
@@ -160,6 +161,10 @@ export const createCompositeBlock = <Options extends CreateCompositeBlockOptions
 
             const nestedBlocks = Object.entries(blockConfigNormalized).filter(([, { nested }]) => nested === true);
 
+            const visibleOrderedBlocks = visibleOrderedBlocksForState?.(state) ?? [];
```

Comment:
This is complicated. Can't you use  `const visibleOrderedBlocks = visibleOrderedBlocksForState?.(state) ?? Object.keys(blocks);` and use visibleOrderedBlocks for rendering, without filter, sort, Set etc.?